src/tabbar.cpp

```diff
@@ -276,6 +278,47 @@ void TabBar::closeTabsByZimId(const QString &id)
     }
 }
 
+int TabBar::indexOfWebView(QObject *object) const
+{
+  for (int index = 0; index <= mp_stackedWidget->count(); index++)
+  {
+    if (ZimView* zv = qobject_cast<ZimView*>(mp_stackedWidget->widget(index)))
+        if (zv->getWebView() == object)
+            return index;
+  }
+  return -1;
+}
+
+void TabBar::changeTabUrlInSave(int index, const QString &url)
+{
+    QStringList tabs = KiwixApp::instance()->getSettingsManager()->getTabList();
+    tabs[index] = url;
+    KiwixApp::instance()->getSettingsManager()->setTabList(tabs);
+}
+
+void TabBar::insertTabToSave(int index, const QString &url)
+{
+    QStringList tabs = KiwixApp::instance()->getSettingsManager()->getTabList();
+    tabs.insert(index, url);
+    KiwixApp::instance()->getSettingsManager()->setTabList(tabs);
+}
+
+void TabBar::removeTabFromSave(int index)
+{
+    QStringList tabs = KiwixApp::instance()->getSettingsManager()->getTabList();
+    tabs.removeAt(index);
+    KiwixApp::instance()->getSettingsManager()->setTabList(tabs);
+}
+
+void TabBar::swapTabInSave(int from, int to)
+{
+    QStringList tabs = KiwixApp::instance()->getSettingsManager()->getTabList();
+    QString tabUrl = tabs[from];
+    tabs.removeAt(from);
+    tabs.insert(to, tabUrl);
+    KiwixApp::instance()->getSettingsManager()->setTabList(tabs);
+}
```

Comment:
Why such a complex scheme of maintaining the list of open tabs? Can't you just generate/save the list of open tabs (by iterating over them) during the termination of the application?

Reply:
Done